src/topology/separation.lean

```diff
@@ -716,15 +716,16 @@ lemma induced_bot {X Y : Type*} {f : X → Y} (hf : function.injective f) :
   topological_space.induced f ⊥ = ⊥ :=
 eq_of_nhds_eq_nhds (by simp [nhds_induced, ← set.image_singleton, hf.preimage_image, nhds_bot])
 
+lemma inducing.discrete_topology {X Y : Type*} [topological_space X] [topological_space Y]
+  [discrete_topology Y] {f : X → Y} (h1 : inducing f) (h2 : function.injective f) :
+  discrete_topology X :=
+⟨by rw [h1.induced, discrete_topology.eq_bot Y, induced_bot h2]⟩
+
 /-- The topology induced under an inclusion `f : X → Y` from the discrete topological space `Y`
 is the discrete topology on `X`. -/
 lemma discrete_topology_induced {X Y : Type*} [tY : topological_space Y] [discrete_topology Y]
-  {f : X → Y} (hf : function.injective f) : @discrete_topology X (topological_space.induced f tY) :=
-begin
-  constructor,
-  rw discrete_topology.eq_bot Y,
-  exact induced_bot hf
-end
+  {f : X → Y} (hf : function.injective f) : @discrete_topology X (induced f tY) :=
+let tX := induced f tY in by exactI inducing.discrete_topology ⟨rfl⟩ hf
```

Comment:
I feel like this lemma should come first and `embedding.discrete_topology` should be proved in terms of it, is it harder in any way?

Reply:
It seems to be a little more annoying, because you would prove `inducing.discrete_topology` by invoking `discrete_topology.eq_bot` on the output of `discrete_topology_induced`, but now you are dealing with two topologies on `X` and have to use a clunky `@`-notation.

Reply:
Oh okay that makes sense